react-components/src/notifications/view/dialogs/JoinTeamRequestDialog.js

```diff
@@ -0,0 +1,304 @@
+        requester_name,
+        team_name
+    } = props;
+    return (
+        <div style={{display: action === DENY ? "block" : "none"}}>
+            <Card className={classes.card} raised={true}>
+                <CardHeader title={getMessage("denyRequestHeader")}/>
+                <CardContent>
+                    <Typography paragraph>
+                        {getMessage("denyRequestMessage", {
+                            values: {
+                                name: requester_name,
+                                team: team_name
+                            }
+                        })}
+                    </Typography>
+                </CardContent>
+                <CardActions>
+                    <TextField
+                        InputLabelProps={{
+                            shrink: true,
+                        }}
+                        placeholder="Message"
+                        fullWidth
+                        margin="normal"
+                        onChange={onChange}
+                    />
+                </CardActions>
+            </Card>
+        </div>
+    );
+}
+
+class JoinTeamRequestDialog extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dialogOpen: props.dialogOpen,
+            action: APPROVE,
+            privilege:"read",
+            message: "",
+            loading: false,
+        };
+        this.handleCancelClick = this.handleCancelClick.bind(this);
+        this.handleOkClick = this.handleOkClick.bind(this);
+    }
+
+    handleCancelClick() {
+        this.setState({dialogOpen: false});
+    }
+
+    handleOkClick() {
+        this.setState({loading: true});
+        if (this.state.action === APPROVE) {
+            this.props.presenter.addMemberWithPrivilege(this.state.privilege, (result) => {
+                this.setState({dialogOpen: false});
+            }, (errorCode, errorMessage) => {
+                this.setState({
+                    dialogOpen: false,
+                    loading: false
+                });
+            });
+        } else {
+            this.props.presenter.denyRequest(this.state.message, (result) => {
+                this.setState({dialogOpen: false});
+            }, (errorCode, errorMessage) => {
+                this.setState({
+                    dialogOpen: false,
+                    loading: false
+                });
+            });
+        }
+    }
+    render() {
+        const classes = this.props.classes;
+        const {
+            requester_name,
+            team_name,
+        } = this.props.request;
+        const {dialogOpen} = this.state;
+        return (
+            <React.Fragment>
+                <Dialog
+                    id={ids.JOIN_REQUEST_DLG}
+                    open={dialogOpen}
+                    onClose={this.props.handleJoinTeamRequestDialogClose}
+                >
+                    <DialogTitle style={{backgroundColor: Color.blue}}>
```

Comment:
Maybe not for this PR, but we have enough dialogs with the same background color, maybe we should separate out our own `DEDialog` component or something so we all use the same style?

Reply:
Already done. Coming up in next PR.